comptime if doesn't work for `const`

```
OS: linux, "Manjaro Linux"
Processor: 32 cpus, 64bit, little endian, AMD Ryzen 9 5950X 16-Core Processor
CC version: cc (GCC) 12.1.0

getwd: /home/jamie
vmodules: /home/jamie/.vmodules
vroot: /home/jamie/git/v
vexe: /home/jamie/git/v/v
vexe mtime: 2022-07-21 14:18:11
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 a46bcf3.be9f8cc

Git version: git version 2.37.1
Git vroot status: weekly.2022.29-20-gbe9f8cc7
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452
```
**What did you do?**
```v
const msg = $if windows { 'windows, eh?' } $else { 'ok, then' }
```
**What did you expect to see?**

No errors
 
**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 000004f5
tcc: error: file 'crt1.o' not found
/tmp/v_1000/foo.14280320083023842857.tmp.c:1929: error: declaration of void object
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1338,6 +1338,15 @@ pub fn (mut c Checker) const_decl(mut node ast.ConstDecl) {
 			}
 		}
 		node.fields[i].typ = ast.mktyp(typ)
+		if mut field.expr is ast.IfExpr {
+			stmts := field.expr.branches[0].stmts
+			if stmts.len > 0 && stmts.last() is ast.ExprStmt
+				&& (stmts.last() as ast.ExprStmt).typ != ast.void_type {
+				field.expr.is_expr = true
+				field.expr.typ = (stmts.last() as ast.ExprStmt).typ
+				field.typ = field.expr.typ
+			}
+		}
 		c.const_deps = []
 		c.const_var = prev_const_var
 	}

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1339,12 +1339,20 @@ pub fn (mut c Checker) const_decl(mut node ast.ConstDecl) {
 		}
 		node.fields[i].typ = ast.mktyp(typ)
 		if mut field.expr is ast.IfExpr {
-			stmts := field.expr.branches[0].stmts
-			if stmts.len > 0 && stmts.last() is ast.ExprStmt
-				&& (stmts.last() as ast.ExprStmt).typ != ast.void_type {
-				field.expr.is_expr = true
-				field.expr.typ = (stmts.last() as ast.ExprStmt).typ
-				field.typ = field.expr.typ
+			if field.expr.branches.len == 2 {
+				first_stmts := field.expr.branches[0].stmts
+				second_stmts := field.expr.branches[1].stmts
+				if first_stmts.len > 0 && first_stmts.last() is ast.ExprStmt
+					&& (first_stmts.last() as ast.ExprStmt).typ != ast.void_type {
+					field.expr.is_expr = true
+					field.expr.typ = (first_stmts.last() as ast.ExprStmt).typ
+					field.typ = field.expr.typ
+				} else if second_stmts.len > 0 && second_stmts.last() is ast.ExprStmt
+					&& (second_stmts.last() as ast.ExprStmt).typ != ast.void_type {
+					field.expr.is_expr = true
+					field.expr.typ = (second_stmts.last() as ast.ExprStmt).typ
+					field.typ = field.expr.typ
+				}
 			}
 		}
 		c.const_deps = []

--- a/vlib/v/checker/if.v
+++ b/vlib/v/checker/if.v
@@ -227,7 +227,7 @@ pub fn (mut c Checker) if_expr(mut node ast.IfExpr) ast.Type {
 					c.error('mismatched types `${c.table.type_to_str(node.typ)}` and `${c.table.type_to_str(last_expr.typ)}`',
 						node.pos)
 				}
-			} else {
+			} else if !node.is_comptime {
 				c.error('`$if_kind` expression requires an expression as the last statement of every branch',
 					branch.pos)
 			}

--- a/vlib/v/tests/comptime_if_expr_in_const_decl_test.v
+++ b/vlib/v/tests/comptime_if_expr_in_const_decl_test.v
@@ -1,0 +1,10 @@
+const msg = $if windows {
+	'windows, eh?'
+} $else {
+	'ok, then'
+}
+
+fn test_comptime_if_in_const_decl() {
+	println(msg)
+	assert msg.len > 0
+}

